src/common/mask/index.ts

```diff
@@ -0,0 +1,93 @@
+export default function (input:HTMLInputElement, initialMask:string) {
```

Comment:
The parts of this logic that handles events should probably move to `phone-input/component.ts`, while this file just works with the strings and _maybe_ with the DOM for cursor selection. Detaching events like this is harder to follow and also more brittle since you're escaping from Marko's event handling.

Reply:
Can you explain what you mean? Not sure I follow.

Reply:
It's not the end of the world if you keep it like this, and if you end up extracting the mask logic to an external module like makeup this is the _only_ way to do it, but if you are only using logic inside Marko it typically makes more sense to use Marko's event handling system instead of `createEventListener`

Reply:
Yea, I agree with Luke here. If ebay ui is to be a Marko component library, it would be great to utilize Marko's event handling instead of native `js`.